Made animations coherent as per iOS spec.

AlexGustafsson opened the pull request:
## Why

The animations for the Navigation Bar were currently triggered after the transition. As described in the source comments, this was not intended behavior. 
## What

Coherent animations were implemented/fixed to enable animations to start at the same time as the transition. Also went from Tween values to Animated as per React Native docs.

AlexGustafsson committed:
Made animations coherent as per iOS spec. Went from Tweening to Animated as per React Native docs.

AlexGustafsson committed:
Corrected error when initializing new animated value

AlexGustafsson committed:
Edited animation duration to depict iOS and not dev animations

MikaelCarpenter commented:
wow looks great man! can you also take `react-tween-state` out of dependencies in `package.json` since it's no longer needed.

AlexGustafsson committed:
Removed react-tween-state from dependencies

Removed dependency of react-tween state since it's no longer needed
(replaced with Animated)

AlexGustafsson commented:
@MikaelCarpenter done and done!
